### ISSUE TITLE
How can I disable lint checks for a code block?

### ISSUE THREAD

**Issue by huangjs** [HairyFotr/linter#19]

**Comment by HairyFotr** [comment]
Currently you cannot, but you can ignore specific lines with //linter:disable[:OptionalCheckNames...]

What kind of syntax would you recommend?

**Comment by huangjs** [comment]
I would recommend scalastyle-like switcher.

// linter:off
...
// linter:on


Jianshi


On Thu, Mar 19, 2015 at 6:33 PM, Matic Potočnik <notifications@github.com>
wrote:

> Currently you cannot, but you can ignore specific lines with
> //linter:disable[:OptionalCheckNames...]
>
> What kind of syntax would you recommend?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HairyFotr/linter/issues/19#issuecomment-83496530>.
>



-- 
Jianshi Huang

LinkedIn: jianshi
Twitter: @jshuang
Github & Blog: http://huangjs.github.com/

**Comment by HairyFotr** [comment]
Yeah, that sounds good, I'll try to implement something over the weekend.

I'll likely also replace "disable" in favour of "ignore" to make it even more similar to scalastyle, and because having "disable" and "off"/"on" together would be confusing.

**Comment by huangjs** [comment]
Thanks!

Jianshi

On Fri, Mar 20, 2015 at 12:51 AM, Matic Potočnik <notifications@github.com>
wrote:

> Yeah, that sounds good, I'll try to implement something over the weekend.
>
> I'll likely also replace "disable" in favour of "ignore" to make it even
> more similar to scalastyle, and because having "disable" and "off"/"on"
> together would be confusing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HairyFotr/linter/issues/19#issuecomment-83659892>.
>



-- 
Jianshi Huang

LinkedIn: jianshi
Twitter: @jshuang
Github & Blog: http://huangjs.github.com/

**Comment by ngbinh** [comment]
any update on this?

**Comment by TomasMikula** [comment]
👍
I would also like to be able to turn off just specific warnings.

**Comment by KANGOD** [comment]
Android Studio (or IntelliJ)'s <kbd>Ctrl</kbd><kbd>Alt</kbd><kbd>L</kbd> formatter supports similar grammar to ignore a code block, ref https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html#exclude_part_of_code

    // @formatter:off
    blahblah
    // @formatter:on

Of course 
```
// linter:disable[:OptionalCheckNames...]
// linter:enable[:OptionalCheckNames...]
```
is appreciate, while if it's difficult to implement, a whole off / on is also welcome.
